src/components/AddApplication/AddApplicationDescription.js

```diff
@@ -140,7 +101,7 @@ AddApplicationDescription.propTypes = {
     })
 };
 
-const mapStateToProps = ({ providers: { entities, appTypes, sourceTypes } }, { match: { params: { id } } }) =>
-    ({ source: entities.find(source => source.id  === id), appTypes, sourceTypes });
+const mapStateToProps = ({ providers: { entities, sourceTypes } }, { match: { params: { id } } }) =>
+    ({ source: entities.find(source => source.id  === id), sourceTypes });
```

Comment:
What happens if there is no source? Since you can write anything into the URL this case should be handled.